core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -718,8 +743,16 @@ public void renameTableTargetAlreadyExistsAsView() {
     assertThat(catalog().viewExists(viewIdentifier)).as("View should exist").isTrue();
 
     assertThatThrownBy(() -> tableCatalog().renameTable(tableIdentifier, viewIdentifier))
-        .isInstanceOf(AlreadyExistsException.class)
-        .hasMessageContaining("Cannot rename ns.table to ns.view. View already exists");
+        .satisfiesAnyOf(
+            throwable ->
+                assertThat(throwable)
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessageContaining("new table ns.view already exists"),
```

Comment:
why is this one needed? I think it might be better to adjust the error reporting on the catalog side

Reply:
It seems all the catalog is not following the same style of error message. 
One example:
[InMemoryCatalog](https://github.com/apache/iceberg/blob/4a3d266a6946f00602cfddc809e5480f3139f352/core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java#L148) has different error message in case of failed validation, while 
[HiveCatalog](https://github.com/apache/iceberg/blob/4a3d266a6946f00602cfddc809e5480f3139f352/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L224)  might throw different error message. 
Only way I can think of here is follow the same error message for all the Catalogs who us trying to use Catalog Level tests. Or add the or clause with applicable error message. 

Please share your thoughts.